python-package/lightgbm/basic.py

```diff
@@ -2363,6 +2368,10 @@ def add_features_from(self, other):
         if other.data is None:
             self.data = None
         elif self.data is not None:
+            if isinstance(self.data, Sequence) or isinstance(other.data, Sequence):
+                raise ValueError('Add features from Sequence is not supported.'
+                                 'Please combine features in your Sequence implementation.')
```

Comment:
If you want to prohibit using `add_features_from` with `Sequence` at all, you need to move this error early than calling combining features at cpp side.
https://github.com/microsoft/LightGBM/blob/5b7a6f3e7150aeb704d1dd2b852d246af3e913a3/python-package/lightgbm/basic.py#L2360

Alternatively, I think it will be enough to yield a default warning 
https://github.com/microsoft/LightGBM/blob/5b7a6f3e7150aeb704d1dd2b852d246af3e913a3/python-package/lightgbm/basic.py#L2419-L2424
which informs users that features were combined at cpp side but they are not allowed to call `get_data()` anymore at Python side.

Reply:
OK. As the cpp side is doing actual combining, I removed the raise statement.